### PR TITLE
Feature/farm organisation

### LINKED ISSUE #86: Remove tau-leaping references
The original code used tau-leaping, however, since the events in the current refactoring happen every day tau-leaping is no longer needed. 

This issue is about removing the leftover references of tau-leaping. from the current code.

### LINKED ISSUE #96: Group farms by organisation
- Organisation have a limited capital, and treatment incurs a cost
- Treatment must be applied when the number of lice reaches a given threshold simultaneously in all farms, not just cages.

Right now, the driver performs bulk actions. This is perfectly okay but some restructuring needs to be done to add some extra information.

Note: there may be multiple organisations in a lake. While it is to be expected lice would move across such farms, synergic behaviour is not to be expected.

Requires #25 to be closed first.

### PR DESCRIPTION
**Pull request by erolm-a**
Closes #86, #96, #105

TODO:

- [X] basic organisation structure + revamp runner
- [X] implement a number of events (fishing, fallowing) and associate a reward/cost to them. For the first, we'd expect the reward to be proportional to a number of factors (number of fish, weight distribution, lice-to-fish ratio etc.). For the second, we need a resume date.
- [X] make scheduling treatment an expensive operation
- [ ] ~~Farm communication with organisation, stub of game theory logic~~ (to do in another PR)
- [x] Tests
    - [X] Cage update cost
    - [x] Farm update cost when applying treatment
    - [x] Treatment limitation
    - [ ] ~~Organisation coordination~~ (to do in another PR)

### CODE REVIEW

**Review by anthonyohare — COMMENTED** [review]
I think the capital should be member of the farm because the organisation can bbbe (and is usually) a more fluid entity; it is really a management agreement between farms.

The step size was always = 1 [day]. Could we possibly leave it and set the step size as a default parameter with a value =1? I'm not sure if this is even a good idea though and making it work correctly may be even more trouble than just taking it out completely.

**Comment by erolm-a** [comment]
Does that mean each farm is going to be an individual agent? To which level is coordination possible?

**Comment by anthonyohare** [comment]
Yes, my understanding is that the farms are individual agents who communication through the organisation. For example, if one farm samples fish and decides that they need to treat they will send a message to the org that they are treating and the org will tell the other farms they must treat.

Note for the near future: when a farm is told it must treat it must decide whether or not to cooperate. If they sample fish and need to treat then they will, else (if the mean numberer of lice per fish on their farm is less that the trigger) then they will cooperate with some probability [and this is a parameter that comprises its strategy].

**Comment by erolm-a** [comment]
For now, this behaviour of the agents will be done in another branch as this branch is getting too big to be maintainable (although many of the modifications are mere refactorings)

File: refactor/config_data/Fyne/params.json [hunk 1]
```diff
@@ -53,11 +57,19 @@
...
+                        "2017-12-01 00:00:00",
+                        "2018-02-01 00:00:00",
+                        "2018-05-01 00:00:00"
                     ]
+                },
+                "start_capital": {
+                    "description": "Organisation capital (in GBP)",
+                    "value": "100000.00"
+                },
+                "max_num_treatments": {
```
Comment (erolm-a) [inline refactor/config_data/Fyne/params.json:70]: Note: this needs to be replaced (or enriched) with a time gap.

**Pull request merged**